core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java

```diff
@@ -0,0 +1,506 @@
+    PowerMockito.mockStatic(UnderFileSystem.Factory.class);
+    Mockito.when(UnderFileSystem.Factory.create(anyString(), any())).thenReturn(mUfs);
+
+    ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    ServerConfiguration.set(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS, mUfsUri);
+    String journalFolderUri = mTempDir.newFolder().getAbsolutePath();
+
+    mExecutorService = Executors
+        .newSingleThreadExecutor(ThreadFactoryUtils.build("DefaultFileSystemMasterTest-%d", true));
+    mRegistry = new MasterRegistry();
+    JournalSystem journalSystem =
+        JournalTestUtils.createJournalSystem(journalFolderUri);
+    CoreMasterContext context = MasterTestUtils.testMasterContext(journalSystem);
+    new MetricsMasterFactory().create(mRegistry, context);
+    BlockMaster blockMaster = new BlockMasterFactory().create(mRegistry, context);
+    mFileSystemMaster = new DefaultFileSystemMaster(blockMaster, context,
+        ExecutorServiceFactories.constantExecutorServiceFactory(mExecutorService));
+    mRegistry.add(FileSystemMaster.class, mFileSystemMaster);
+    journalSystem.start();
+    journalSystem.gainPrimacy();
+    mRegistry.start(true);
+
+    MetricsSystem.resetAllMetrics();
+  }
+
+  @After
+  public void after() throws Exception {
+    mRegistry.stop();
+    mExecutorService.shutdown();
+  }
+
+  @Test
+  public void testMetadataSyncMetrics() throws Exception {
+    final Counter count = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_COUNT;
+    final Counter success = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_SUCCESS;
+    final Counter fail = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_FAIL;
+    final Counter noChange = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_NO_CHANGE;
+    final Counter skipped = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_SKIPPED;
+    final Counter pathSuccess =
+        DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_SYNC_PATHS_SUCCESS;
+    final Counter pathFail = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_SYNC_PATHS_FAIL;
+    final Counter timeMS = DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_TIME_MS;
+
+    createDir("/dir0");
+    OutputStream outputStream = createFile("/dir0/file0");
+    outputStream.write(new byte[] {0});
+    outputStream.close();
+
+    //path is non-existent in inodeTree,existent in UFS and UFS is available ->success and no change
+    LockingScheme syncScheme =
+        new LockingScheme(new AlluxioURI("/"), InodeTree.LockPattern.READ, true);
+    InodeSyncStream inodeSyncStream =
+        new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+            DescendantType.ALL, ListStatusContext.defaults().getOptions().getCommonOptions(),
+            false, true, false, false);
+    inodeSyncStream.sync();
+    assertEquals(0 + 1, count.getCount());
+    assertEquals(0 + 1, success.getCount());
+    assertEquals(0 + 3, pathSuccess.getCount());
```

Comment:
what are the 3 paths? `/`, `/dir0` and `/dir0/file0`?

Reply:
For lines like this, if the number change is not straightforward, pls add a comment explaining where the changes are from. You don't need to explain all numbers, just the ones that are not trivial to understand.